### ISSUE TITLE
Add support for import.meta.env

### ISSUE THREAD

**Issue by brielov** [jpruliere/env-autocomplete#12]
Some new bundlers like [vite](https://vitejs.dev/guide/env-and-mode.html) make use of `import.meta` API to load environment variables. It would be nice to have support for it.

**Comment by jpruliere** [comment]
Hi @brielov !

Sorry for the eternity it took me to answer you, I am pretty busy now and unfortunately, I can't handle any further development of this extension. But in a near future, I will transfer ownership to @maxcutlyp who wants to be involved in this extension development, so stay tuned 😉 

Manually integrating each and every bundler's "fantasy" does not look like a sane plan to me. Especially since I read in vite's doc that `.env` are also parsed, so I would strongly recommend keeping your app "untangled" by using more standard ways of passing env vars to Node. By untangled, I mean you might consider moving to another bundler someday (I am a backend dev, I don't bundle much so I don't know if this kind of thing happens or not).

Anyway, I can think of a way to make everyone happy : configuration.

VScode extensions can easily register config fields in the Settings page, you could pass some extra files or glob patterns you would like to include in autocompletion. It is no small development though, but it could be worth a shot.

**Comment by maxcutlyp** [comment]
To be fair, from some cursory googling it seems like `import.meta.env` isn't just a Vite thing, but more an ES modules thing. For example, it's used in other bundlers like Snowpack, and [webpack dropped support for the `process` keyword entirely in v5](https://webpack.js.org/migrate/5/#run-a-single-build-and-follow-advice). That said, a lot of discourse about it revolves around or mentions Vite, but I still think it would be worth adding support for it explicitly for these reasons.

However, upon doing some local testing and subsequent googling, it turns out that VSCode extensions cannot use the `import` keyword at all (which is typically used as a replacement for `require`, a la `import method from 'package'`, but is also used for `import.meta.env`) because of a limitation in VSCode itself, which is potentially because of a limitation in Electron - see microsoft/vscode#130367 and electron/electron#21457.

There is also the issue that, since `import.meta.env` is implemented by various different bundlers (noteably: none of which are used by this extension), we would not be able to generate completions that are not in a `.env` file the way we do with `process.env` - that is, by borrowing from the extension's own `process.env` - since the extension does not have an `import.meta.env` to borrow from.

We could "fake it" to a degree by just using `process.env` for both, but I'm not sure that they necessarily share all the same values, and in my opinion, showing an incomplete/inaccurate list is worse than showing nothing at all, as the former implies that it is complete/accurate and would cause confusion if it was not, while the latter implies that it is simply unsupported, which is true.

Another option would be to provide only completions from `.env` files for `import.meta.env`, but this would arguably still run into the same issue.

TLDR: unfortunately, I don't think reliable support for `import.meta.env` will be possible at this stage, but I'm open to suggestions here so I won't close the issue yet.

**Comment by maxcutlyp** [comment]
Closed as there's not really anything new here. Feel free to open it again if you think there's something worth opening it for.
